homeassistant/components/camera/__init__.py

```diff
@@ -0,0 +1,1133 @@
+                       ATTR_DISCOVERED: data})
+
+        entity.check_required_configurators()
+
+    # -------------------------------------------------------------------------
+    # CAMERA COMPONENT ENDPOINTS
+    # -------------------------------------------------------------------------
+    # The following defines the endpoints for serving images from the camera
+    # via the HA http server.  This is means that you can access images from
+    # your camera outside of your LAN without the need for port forwards etc.
+
+    # Because the authentication header can't be added in image requests these
+    # endpoints are secured with session based security.
+
+    # pylint: disable=unused-argument
+    def _proxy_camera_image(handler, path_match, data):
+        """ Proxies the camera image via the HA server. """
+        entity_id = path_match.group('entity_id')
+
+        camera = None
+        if entity_id in component.entities.keys():
+            camera = component.entities[entity_id]
+
+        if camera:
+            response = camera.get_camera_image()
+            handler.wfile.write(response.content)
+        else:
+            handler.send_response(HTTP_NOT_FOUND)
+
+    hass.http.register_path(
+        'GET',
+        re.compile(r'/api/camera_proxy/(?P<entity_id>[a-zA-Z\._0-9]+)'),
+        _proxy_camera_image,
+        require_auth=True)
+
+    # pylint: disable=unused-argument
+    def _proxy_camera_mjpeg_stream(handler, path_match, data):
+        """ Proxies the camera image as an mjpeg stream via the HA server.
+        This function takes still images from the IP camera and turns them
+        into an MJPEG stream.  This means that HA can return a live video
+        stream even with only a still image URL available.
+        """
+        entity_id = path_match.group('entity_id')
+
+        camera = None
+        if entity_id in component.entities.keys():
+            camera = component.entities[entity_id]
+
+        if camera:
+            camera.last_connected_address = handler.address_string()
+            message = "{0} started streaming to {1}".format(
+                camera.name, handler.address_string())
+
+            hass.bus.fire(
+                "camera_stream_started",
+                {
+                    "component": DOMAIN,
+                    ATTR_ENTITY_ID: entity_id,
+                    ATTR_FRIENDLY_LOG_MESSAGE: message})
+
+            try:
+                camera.is_streaming = True
+                camera.update_ha_state()
+
+                http = urllib3.PoolManager()
+                handler.request.sendall(bytes('HTTP/1.1 200 OK\r\n', 'utf-8'))
+                handler.request.sendall(bytes(
+                    'Content-type: multipart/x-mixed-replace; \
+                        boundary=--jpgboundary\r\n\r\n', 'utf-8'))
+
+                handler.request.sendall(bytes('--jpgboundary\r\n', 'utf-8'))
+
+                while True:
+                    headers = urllib3.util.make_headers()
```

Comment:
It seems that this section is the only place where `urllib3` is being used. I propose we rewrite this part to use the existing dependency `requests`.

```python
import requests
from requests.auth import HTTPBasicAuth

if camer.username and camera.password:
  response = requests.get(camera.still_image_url, auth=HTTPBasicAuth(camera.username, camera.password))
else:
  response = requests.get(camera.still_image_url)

# Use [response.content](http://docs.python-requests.org/en/latest/user/quickstart/#binary-response-content) instead of req.data
```

(disclaimer: I did not test this code)